### PR TITLE
librewolf-bin: init at 89.0.2-1

### PR DESCRIPTION
**Pull request by dan4ik605743**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/librewolf-bin/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/librewolf
+    '';
+
+  meta = with lib; {
+    description = "A fork of Firefox, focused on privacy, security and freedom.";
+    homepage = "https://librewolf-community.gitlab.io/";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ dan4ik605743 ];
+    platforms = platforms.linux;
```
Comment (L-as) [inline pkgs/applications/networking/browsers/librewolf-bin/default.nix:39]: This is incorrect. The binary build only supports x86_64-linux.

Comment (dan4ik605743) [reply pkgs/applications/networking/browsers/librewolf-bin/default.nix:39]: platform.isAarch64
Right?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129081` at 08ca5f98 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>librewolf-bin</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `makeDesktopItem`.
Near pkgs/applications/networking/browsers/librewolf-bin/default.nix:2:3:
```
  |
2 | , makeDesktopItem
  |   ^
```
</li>
  </ul>
</details>

**Comment by Kranzes** [comment]
Do you think it will be possible to use something other than the appimage? maybe even build it from source?

**Comment by dan4ik605743** [comment]
> Do you think it will be possible to use something other than the appimage? maybe even build it from source?

The problem is in the glibc version. In NixOS 2-32. He also uses 2-33

**Comment by Kranzes** [comment]
> > Do you think it will be possible to use something other than the appimage? maybe even build it from source?
> 
> The problem is in the glibc version. In NixOS 2-32. He also uses 2-33

Sorry i did not quite get that, could you reword that sentence? :smiley:

**Comment by Kranzes** [comment]
Are you saying that in order to build it from source we need Glibc 2.33? Why wont we start by updating Glibc then?

**Comment by dan4ik605743** [comment]
> Are you saying that in order to build it from source we need Glibc 2.33? Why wont we start by updating Glibc then?

Not even for source. And just for the binary. Yes, you need glibc-2.33

**Comment by Kranzes** [comment]
Then i think we should focus on getting glibc 2.33 up and running instead of cluttering nixpkgs with more appimages.

**Comment by dan4ik605743** [comment]
> Then i think we should focus on getting glibc 2.33 up and running instead of cluttering nixpkgs with more appimages.

Oh well. There is another detail. A couple of versions below the new glibc is not needed. You can push it to nixpkgs for now. How do you like the idea?

**Comment by Kranzes** [comment]
I personally prefer that, I am not a fan of appimages myself.

**Comment by dan4ik605743** [comment]
> I personally prefer that, I am not a fan of appimages myself.

Okay. Now I'll fill in another build with a lower version. When will glibc. I will update it.

**Comment by dan4ik605743** [comment]
About appimage. I don’t know .. As for me, not a very bad option, given this situation

**Comment by Kranzes** [comment]
Though updating glibc can be a ton of work and will take years to evaluate due to the fact that every package that uses stdenv depends on it.

**Comment by dan4ik605743** [comment]
I'm talking about this

**Comment by dan4ik605743** [comment]
From him all the threads

**Comment by dan4ik605743** [comment]
Maybe this package is still valid?

**Comment by Kranzes** [comment]
I actually am not sure, I think we should get more people to discuss whether it is valid or not.

**Comment by dan4ik605743** [comment]
> I actually am not sure, I think we should get more people to discuss whether it is valid or not.

Well, it's up to you. My job is to propose. I tested the package, everything works fine. This will make it easier to update it. I just need to correct the platforms and remove the unused-argument. And the build is ready

**Comment by Kranzes** [comment]
@dan4ik605743 take a look at this https://github.com/NixOS/nixpkgs/pull/111616 :)

**Comment by L-as** [comment]
@dan4ik605743 I'm asking you since I couldn't find much information about it, but if librewolf is just firefox with patches, couldn't we just copy the firefox derivation and change the sources, etc.?

File: pkgs/applications/networking/browsers/librewolf-bin/default.nix (hunk 1 above)
Comment (L-as) [reply pkgs/applications/networking/browsers/librewolf-bin/default.nix:39]: just `platforms = [ "x86_64-linux" ]` *should* be fine?

**Comment by L-as** [comment]
see https://gitlab.com/librewolf-community/browser/arch/-/blob/master/PKGBUILD, I am against merging this. I think we should build it ourselves. It does not seem very hard.

**Comment by Kranzes** [comment]
I agree!

**Comment by dan4ik605743** [comment]
That is, the decision has been made to collect it from sources?

**Comment by L-as** [comment]
Yes, it should not be very hard.

**Comment by dan4ik605743** [comment]
> Yes, it should not be very hard.

Okay. I just thought, let's say I write a build. Here's how to test it .. I have an i3-5005U. And the browser will collect from the resources.

**Comment by dan4ik605743** [comment]
Immediately if tested from PR.Immediately if you test from PR. But I don't poke anything myself

**Comment by Kranzes** [comment]
> > Yes, it should not be very hard.
> 
> Okay. I just thought, let's say I write a build. Here's how to test it .. I have an i3-5005U. And the browser will collect from the resources.

I will be able to build it with my R7 2700x and I am pretty sure that ofborg can also eval and build it.

**Comment by dan4ik605743** [comment]
> > > Yes, it should not be very hard.
> > 
> > 
> > Okay. I just thought, let's say I write a build. Here's how to test it .. I have an i3-5005U. And the browser will collect from the resources.
> 
> I will be able to build it with my R7 2700x and I am pretty sure that ofborg can also eval and build it.

Okay, I'll get to the build then.

**Comment by Kranzes** [comment]
Also where did you see that librefox requires glibc 2.33? wouldnt that mean firefox also needs it?

**Comment by Kranzes** [comment]
Also where did you see that librefox requires glibc 2.33? 

> see https://gitlab.com/librewolf-community/browser/arch/-/blob/master/PKGBUILD, I am against merging this. I think we should build it ourselves. It does not seem very hard.

The same should also be done for GNU Icecat which i saw that @dan4ik605743 has also made a PR for.

**Comment by dan4ik605743** [comment]
> Also where did you see that librefox requires glibc 2.33?
> 
> > see https://gitlab.com/librewolf-community/browser/arch/-/blob/master/PKGBUILD, I am against merging this. I think we should build it ourselves. It does not seem very hard.
> 
> The same should also be done for GNU Icecat which i saw that @dan4ik605743 has also made a PR for.

**Comment by dan4ik605743** [comment]
oh what have I done. accidentally pushed in the wrong place. sorry

**Comment by dan4ik605743** [comment]
> Also where did you see that librefox requires glibc 2.33?
> 
> > see https://gitlab.com/librewolf-community/browser/arch/-/blob/master/PKGBUILD, I am against merging this. I think we should build it ourselves. It does not seem very hard.
> 
> The same should also be done for GNU Icecat which i saw that @dan4ik605743 has also made a PR for.

The binary demanded it

**Comment by dan4ik605743** [comment]
> Also where did you see that librefox requires glibc 2.33?
> 
> > see https://gitlab.com/librewolf-community/browser/arch/-/blob/master/PKGBUILD, I am against merging this. I think we should build it ourselves. It does not seem very hard.
> 
> The same should also be done for GNU Icecat which i saw that @dan4ik605743 has also made a PR for.

And then why. There it is normally made from a binary. Not appimage. autopatchelfhook successfully completed

**Comment by dan4ik605743** [comment]
Assembly is needed here. And everything is fine there in terms of the binary

File: pkgs/applications/networking/browsers/librewolf-bin/default.nix (hunk 1 above)
Comment (dan4ik605743) [reply pkgs/applications/networking/browsers/librewolf-bin/default.nix:39]: Thank you. I'll try!

**Comment by Kranzes** [comment]
In nixpkgs it is preferable to build the package from source, rather than using the binary that it also ships with.
Nix/NixOS is a source based package manager/distribution, that just happens to have a binary cache that caches the results of the binaries that Hydra compiled from source.

**Comment by dan4ik605743** [comment]
> In nixpkgs it is preferable to build the package from source, rather than using the prebuilt-binary that it also ships with.
> Nix/NixOS is a source based package manager/distribution (just like gentoo), that just happens to have a binary cache that caches the results of the binaries that Hydra compiled from source.

I.e. There is a possibility of assembly-collect. No-binary, appimage, etc. Did I understand correctly?

**Comment by Kranzes** [comment]
I still think we should provide both librewolf and librewolf-bin (same with icecat) just like with firefox.

**Comment by dan4ik605743** [comment]
> I still think we should provide both librewolf and librewolf-bin (same with icecat) just like with firefox.

I completely agree with this. Then this build will eventually run for binary librewolf?

**Comment by Kranzes** [comment]
I meant that in nixpkgs we prefer a build from source over a prebuilt-binary, though we do provide both a lot of the times.

**Comment by dan4ik605743** [comment]
> I meant that in nixpkgs we prefer a build from source over a prebuilt-binary, though we do provide both a lot of the times.

Well, I figured it out. So should I leave this build? For binary librevolf.

**Comment by Kranzes** [comment]
What we want is `librewolf` which is built from source with nix and `librewolf-bin` which is the packaged prebuilt librewolf binary.

**Comment by dan4ik605743** [comment]
> What we want is `librewolf` which is built from source with nix and `librewolf-bin` which is the packaged prebuilt librewolf binary.

I understand. I then make 2 commits from the resources. And immediately 2 packages will get right?

**Comment by Kranzes** [comment]
You can add/update multiple packages to nixpkgs in a single PR, if that is what you are asking.

**Comment by dan4ik605743** [comment]
> You can add/update multiple packages to nixpkgs in a single PR, if that is what you are asking.

No, I know that. I'm asking. I will issue a second commit with a non-binary librewolf. As a result, 2 builds librewolf-bin and librewolf will get into the turnips. I understand correctly?

**Comment by Kranzes** [comment]
Sorry, I didnt understand what you were trying to say.

**Comment by dan4ik605743** [comment]
> Sorry, I didnt understand what you were trying to say.

My build with appimage will end up in nixpkgs as librewolf-bin. Right?

**Comment by Kranzes** [comment]
No, we want to stay away from appimages as much as possible. Take a look at librewolf and librewolf-bin in the AUR.
 If i remember correctly they also dont use any appimages for librewolf.

**Comment by Kranzes** [comment]
Straight from Librewolf's website
image

**Comment by dan4ik605743** [comment]
> No, we want to stay away from appimages as much as possible. Take a look at librewolf and librewolf-bin in the AUR.
> If i remember correctly they also dont use any appimages for librewolf.

Well, I repeat, you need glibc-2.33 there!

**Comment by dan4ik605743** [comment]
image

**Comment by dan4ik605743** [comment]
Arch has glibc-2.33. We do not have it. What other options besides appimage?

**Comment by Kranzes** [comment]
Why would the patches of librewolf make the entire thing require glibc 2.33 out of the sudden? while the latest firefox still does not require it. also take a look at this: https://gitlab.com/librewolf-community/browser/common

**Comment by Kranzes** [comment]
I think if we seriously need 2.33 and it is very hardcoded in there, we should just wait for this PR to be merged: https://github.com/NixOS/nixpkgs/pull/111616

**Comment by dan4ik605743** [comment]
> Why would the patches of librewolf make the entire thing require glibc 2.33 out of the sudden? while the latest firefox still does not require it. also take a look at this: https://gitlab.com/librewolf-community/browser/common

I repeat 
https://gitlab.com/librewolf-community/browser/gentoo/-/blob/master/www-client/librewolf-bin/librewolf-bin-88.0.1-r2.ebuild#L49

**Comment by dan4ik605743** [comment]
> I think if we seriously need 2.33 and it is very hardcoded in there, we should just wait for this PR to be merged: #111616

So that's great. Then we'll wait. Or appimage

**Comment by Kranzes** [comment]
I think we should wait, im very against appimages.

**Comment by Kranzes** [comment]
In the meantime you can either close this PR or make it a draft.

**Comment by dan4ik605743** [comment]
> I think we should wait, im very against appimages.



> I think we should wait, im very against appimages.

Yes, I'm also for, wait. Glibc will exit. I will make a binary from a deb package

**Comment by dan4ik605743** [comment]
> In the meantime you can either close this PR or make it a draft.

Yes, I will. I'll be watching glibc. How to push. I'll start working. And I will do PR. With 2 pkgs.

**Comment by Kranzes** [comment]
why from deb? i think the provide their own prebuilt binary.

**Comment by Kranzes** [comment]
Try to figure out where they get the binary from in the librewolf-bin package in the AUR.

**Comment by dan4ik605743** [comment]
> Try to figure out where they get the binary from in the librewolf-bin package in the AUR.

There is a special archive for arch. Well done through it.

**Comment by dan4ik605743** [comment]
I also tested glibc through it.

**Comment by Kranzes** [comment]
Im a bit lost then, need more people's thoughts on this.

**Comment by gi-yt** [comment]
The version of glibc in debian is 2.31-12. Isn't it possible to just extract the contents from the deb like how it is for the build of opera on nixpkgs ?

**Comment by dan4ik605743** [comment]
> The version of glibc in debian is 2.31-12. Isn't it possible to just extract the contents from the deb like how it is for the build of opera on nixpkgs ?

And so everything is extracted. It's completely different here

**Comment by Kranzes** [comment]
> > I think if we seriously need 2.33 and it is very hardcoded in there, we should just wait for this PR to be merged: #111616
> 
> So that's great. Then we'll wait. Or appimage

Time is up!

**Comment by dan4ik605743** [comment]
> Time is up! I just now noticed that 2.33 is in unstable, not sure for how long that has been the case.

yes, I see it. I'll even tell you so I managed to get acquainted with him ( https://github.com/NixOS/nixpkgs/issues/131168 )

**Comment by dan4ik605743** [comment]
> > > I think if we seriously need 2.33 and it is very hardcoded in there, we should just wait for this PR to be merged: #111616
> > 
> > 
> > So that's great. Then we'll wait. Or appimage
> 
> Time is up! I just now noticed that 2.33 is in unstable, not sure for how long that has been the case.

@Kranzes 
image
We need gcc-unwrapped version 11. We'll wait for him. And everything will work. The error due to glibc has disappeared.

**Pull request closed without merging**